scipy/stats/_stats_py.py

```diff
@@ -1402,20 +1402,27 @@ def describe(a, axis=0, ddof=1, bias=True, nan_policy='propagate'):
                    skewness=array([0., 0.]), kurtosis=array([-2., -2.]))
 
     """
-    a, axis = _chk_asarray(a, axis)
+    xp = array_namespace(a)
+    a, axis = _chk_asarray(a, axis, xp=xp)
 
     contains_nan, nan_policy = _contains_nan(a, nan_policy)
 
     if contains_nan and nan_policy == 'omit':
-        a = ma.masked_invalid(a)
-        return mstats_basic.describe(a, axis, ddof, bias)
+        if is_numpy(xp):
+            a = ma.masked_invalid(a)
+            return mstats_basic.describe(a, axis, ddof, bias)
+        else:
+            msg = ("Use of `nan_policy` and `keepdims` "
+                   "is incompatible with non-NumPy arrays.")
+            raise NotImplementedError(msg)
```

Comment:
Can you add test coverage for this branch, checking locally it doesn't appear that this is hit

Reply:
Oops, turns out this is covered by `_contains_nan` and doesn't need to appear in functions individually. 

While testing locally, I see that gh-20467 caused a failure and worked around it with `_array_api.size`.